New package: wired-notify-0.10.2

fccapria opened the pull request:
<!-- Uncomment relevant sections and delete options which are not applicable -->

#### Testing the changes
- I tested the changes in this PR: **YES**


#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**


<!-- Note: If the build is likely to take more than 2 hours, please add ci skip tag as described in
https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#continuous-integration
and test at least one native build and, if supported, at least one cross build.
Ignore this section if this PR is not skipping CI.
-->

#### Local build testing
- I built this PR locally for my native architecture, (x86_64)

abenson commented:
Please don't open/close PRs, it makes it harder for us to track them.

abenson commented on srcpkgs/wired-notify/template:
```diff
@@ -0,0 +1,20 @@
+pkgname=wired-notify
+version=0.10.2
+revision=1
+archs="i686 x86_64"
```
Why is this restricted?

abenson commented on srcpkgs/wired-notify/template:
```diff
@@ -0,0 +1,20 @@
+archs="i686 x86_64"
+build_style=cargo
+hostmakedepends="pkg-config"
+makedepends="rust cargo pango-devel libXScrnSaver-devel libglib-devel"
```
`rust` and `cargo` would go in `hostmakedepends`, but they are implied by `build_style=cargo`

abenson commented on srcpkgs/wired-notify/template:
```diff
@@ -0,0 +1,20 @@
+build_style=cargo
+hostmakedepends="pkg-config"
+makedepends="rust cargo pango-devel libXScrnSaver-devel libglib-devel"
+depends="dbus xorg cairo pango"
```
These `depends` will be automatically detected, if needed.

abenson commented on srcpkgs/wired-notify/template:
```diff
@@ -0,0 +1,20 @@
+maintainer="Francesco Carmelo Capria <francesco@capria.eu>"
+license="MIT"
+homepage="https://github.com/Toqozz/wired-notify"
+#changelog=""
```
If there's no changelog, just remove the whole line.

abenson commented:
When you make the requested changes, you can use `git commit --amend` to apply to the existing commit, and then `git push -f` to update the PR.

fccapria commented:
> Please don't open/close PRs, it makes it harder for us to track them.

Sorry, I created the first branch with an incorrect name.

fccapria commented:
Anyway thanks for all these reviews.
I will correct all these points, do another build and if successful I will update the PR.

fccapria commented:
> > Hope this isn't a problem.
> 
> It should not be an issue because you can do an interactive rebase and squash all of the commits into the first one, then force push. You should end up with a single commit with the message "New package: wired-notify-0.10.2". Commands are `git rebase -i master` and `git push -f`.

Thank you very much, you have been very helpful!
I just rebased and reduced everything to a commit.
The only small thing is that I made a merge before by mistake and then all the commits in that merge were inserted in the first one.
I hope it's all fine now.

fccapria commented:
> Uh... no. You should get rid of those changes and only keep your changes.

Yes, unfortunately I realized this after I actually did a very stupid thing..
Having said that, I think I have solved!
I'm sorry that I created an absurd fuss for such a simple thing, but first I panicked and did a merge that ruined everything.
Fortunately, after several rebases I have fixed it, I think.

classabbyamp commented on srcpkgs/wired-notify/template:
```diff
@@ -0,0 +1,17 @@
+checksum=7e1c522451887af67f76ddc3022ef26be918e01c44cec10c09d39cfa04b9785e
+
+post_install() {
+	       vlicense LICENSE
```
```suggestion
	vlicense LICENSE
```

fccapria commented:
I added the specific architectures as at the moment the software should only be available for these two.

classabbyamp commented:
why?

fccapria commented:
> why? what prevents it from running on arm/ppc/etc?

Checking the various build templates for other distros of the same package only the x86_64 and i686 architectures are specified.
I haven't tried to build for the other architectures myself but the build checks that GitHub did have failed so I thought I'd specify these two architectures.
Can I remove the specified architectures even if the build fails?

classabbyamp commented:
many other distros don't support the architectures void supports.

the build only failed on 32-bit architectures (i686, armv6l, and armv7l).

[the error](https://github.com/void-linux/void-packages/actions/runs/3191379385/jobs/5210738618#step:7:406) was caused by the upstream not ensuring their types matched, i.e. they were not checking if [this `u64`](https://github.com/Toqozz/wired-notify/blob/4bb3dfff38934c47be818f497af65d1e3a01311c/src/config.rs#L120) and [this `c_ulong`](https://docs.rs/x11/2.20.0/x11/xss/struct.XScreenSaverInfo.html#structfield.idle) could actually be compared ([`c_ulong`](https://doc.rust-lang.org/nightly/std/os/raw/type.c_ulong.html) can be `u32` or `u64`). there may be other errors (because the devs obviously aren't testing anything on 32-bit systems), but this one can be fixed by (or something similar):

```diff
- if info.idle / 1000 >= threshold {
+ if (info.idle as u64) / 1000 >= threshold {
```

fccapria commented:
> many other distros don't support the architectures void supports.

Wow, i get it.

> the build only failed on 32-bit architectures (i686, armv6l, and armv7l).
> 
> [the error](https://github.com/void-linux/void-packages/actions/runs/3191379385/jobs/5210738618#step:7:406) was caused by the upstream not ensuring their types matched, i.e. they were not checking if [this `u64`](https://github.com/Toqozz/wired-notify/blob/4bb3dfff38934c47be818f497af65d1e3a01311c/src/config.rs#L120) and [this `c_ulong`](https://docs.rs/x11/2.20.0/x11/xss/struct.XScreenSaverInfo.html#structfield.idle) could actually be compared ([`c_ulong`](https://doc.rust-lang.org/nightly/std/os/raw/type.c_ulong.html) can be `u32` or `u64`). there may be other errors (because the devs obviously aren't testing anything on 32-bit systems), but this one can be fixed by (or something similar):
> 
> ```diff
> - if info.idle / 1000 >= threshold {
> + if (info.idle as u64) / 1000 >= threshold {
> ```

Yes, I noticed that unfortunately only the 32-bit build was not successful .. in this case what should I do?  Create a patch to apply to the source to fix the problem?

classabbyamp commented:
yes, i would suggest patching this and making a pull request upstream as well with the fixes

fccapria commented:
Okay perfect, thanks a lot!  I'll try to solve and then create another pull request.  In the meantime, do I leave this pull request active or do I close it?

classabbyamp commented:
leave it open

fccapria commented:
Hi, I just updated the commit with the patch for 32-bit architectures.

classabbyamp commented:
I see this message in do_install:
```
warning: package `crossbeam-utils v0.8.5` in Cargo.lock is yanked in registry `crates-io`
```
let's import [this patch](https://github.com/Toqozz/wired-notify/commit/a3211fea137754fcde216fc035ab8321f9a484b9.patch)

fccapria commented:
Hi, I didn't really see that warning, thanks!
Anyway, I just added the new patch and modified the one for 32 bit architectures because it brought a warning during the patching phase.

classabbyamp commented on srcpkgs/wired-notify/patches/32bit.patch:
```diff
@@ -0,0 +1,18 @@
+                             self.layout_windows
+                                 .values_mut()
+                                 .flatten()
+				 .for_each(|w| w.update_mode = UpdateModes::DRAW);
+                        }
+                    },
+                    Err(e) => eprintln!("{}", e),
+                }
+            }
+        }	
```
```suggestion
```
I don't think this section is necessary for patching properly

fccapria replied:
In truth, even in my opinion it is not necessary ..
The only reason I inserted those extra lines is because during the patch phase I was getting two errors (both `patch unexpectedly ends in middle of line`).
By inserting those lines I should have deleted one, although the other still remains.
If you think I can delete them, I change them immediately, also because without all those superfluous lines the patch is much cleaner.

classabbyamp replied:
hmm *removing* these lines removed that warning for me

fccapria replied:
Ah, wow, I remove them instantly!

classabbyamp replied:
ok, I think I know what's going on. your editor isn't adding a newline at the end of the file, so it's not the exact same as what's in the file its patching. `echo "" >> srcpkgs/wired-notify/patches/32bit.patch` should fix it now I think

fccapria replied:
Oh, I get it.
So as soon as the check builds finish I add the newline.
